charts/ColorScale.ts

```diff
@@ -212,38 +240,36 @@ export class ColorScale {
         return colors
     }
 
-    @computed private get numAutoBins(): number {
-        return 5
+    @computed get numAutoBins(): number {
+        return defaultTo(this.config.binningStrategyBinCount, 5)
+    }
+
+    @computed get isManualBuckets(): boolean {
+        return this.config.binningStrategy === ColorScaleBinningStrategy.manual
     }
 
     @computed get numBins(): number {
-        return this.config.isManualBuckets
-            ? this.colorSchemeValues.length
+        return this.isManualBuckets
+            ? this.customNumericValues.length
             : this.numAutoBins
     }
 
-    @computed get binStepSizeDefault(): number {
-        const { numAutoBins, minBinValue, valuesWithoutOutliers } = this
-        if (!valuesWithoutOutliers.length) return 10
+    @computed get binStepSize(): number {
+        const {
+            numAutoBins,
+            minBinValue,
+            sortedNumericValuesWithoutOutliers
+        } = this
+        if (!sortedNumericValuesWithoutOutliers.length) return 10
 
         const stepSizeInitial =
-            (valuesWithoutOutliers[valuesWithoutOutliers.length - 1] -
-                minBinValue) /
+            (last(sortedNumericValuesWithoutOutliers)! - minBinValue) /
```

Comment:
I guess we also need some handling for empty `sortedNumericValuesWithoutOutliers` here?